dlp/snippets/inspect_content.py

```diff
@@ -310,6 +310,102 @@ def inspect_table(
 
 # [END dlp_inspect_table]
 
+
+# [START dlp_inspect_column_values_w_custom_hotwords]
+from typing import Dict, List, Union  # noqa: E402, I100
+
+import google.cloud.dlp  # noqa: F811, E402
+
+
+def inspect_column_values_w_custom_hotwords(
+    project: str,
+    table_data: Dict[str, Union[List[str], List[List[str]]]],
+    info_types: List[str],
+    custom_hotword: str,
+) -> None:
+    """ Uses the Data Loss Prevention API to inspect table data using builtin
+    infoType detector and to exclude entire column of data from inspection
+    results when matched with custom hot-word.
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        table_data: Dictionary representing table data.
+        info_types: The infoType for which hot-word rule is applied.
+        custom_hotword: The custom regular expression used for likelihood boosting.
+    """
+
+    # Instantiate a client
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Construct the `table`. For more details on the table schema, please see
+    # https://cloud.google.com/dlp/docs/reference/rest/v2/ContentItem#Table
+    headers = [{"name": val} for val in table_data["header"]]
+    rows = []
+    for row in table_data["rows"]:
+        rows.append(
+            {"values": [{"string_value": cell_val} for cell_val in row]}
+        )
+    table = {"headers": headers, "rows": rows}
+
+    # Construct the `item` for table to be inspected.
+    item = {"table": table}
+
+    # Prepare info_types by converting the list of strings into a list of
+    # dictionaries.
+    info_types = [{"name": info_type} for info_type in info_types]
+
+    # Construct a rule set with caller provided hot-word, with a likelihood
+    # boost to VERY_UNLIKELY when the hot-word are present
+    hotword_rule = {
+        "hotword_regex": {"pattern": custom_hotword},
+        "likelihood_adjustment": {
+            "fixed_likelihood": google.cloud.dlp_v2.Likelihood.VERY_UNLIKELY
+        },
+        "proximity": {"window_before": 1},
+    }
+
+    rule_set = [
+        {
+            "info_types": info_types,
+            "rules": [{"hotword_rule": hotword_rule}],
+        }
+    ]
+
+    # Construct the configuration dictionary, which defines the entire inspect content task.
+    inspect_config = {
+        "info_types": info_types,
+        "rule_set": rule_set,
+        "min_likelihood": google.cloud.dlp_v2.Likelihood.POSSIBLE,
+        "include_quote": True,
+    }
+
+    # Convert the project id into a full resource id.
+    parent = f"projects/{project}"
```

Comment:
this uses the implied global location. it is better to be explicit.

```suggestion
    parent = f"projects/{project}/locations/global"
```